scripts/clean-shrinkwrap.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * This script is in charge of generating the `shrinkwrap` file.
+ *
+ * `npm shrinkwrap` has a bug where it will add optional dependencies
+ * to `npm-shrinkwrap.json`, therefore causing errors if these optional
+ * dependendencies are platform dependent and you then try to build
+ * the project in another platform.
+ *
+ * As a workaround, we keep a list of platform dependent dependencies in
+ * the `platformSpecificDependencies` property of `package.json`,
+ * and manually remove them from `npm-shrinkwrap.json` if they exist.
+ *
+ * See: https://github.com/npm/npm/issues/2679
+ */
+
+/* eslint-disable lodash/prefer-lodash-method */
+/* eslint-disable no-magic-numbers */
+/* eslint-disable no-undefined */
+
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const _ = require('lodash')
+
+const JSON_INDENT = 2
+const SHRINKWRAP_FILENAME = path.join(__dirname, '..', 'npm-shrinkwrap.json')
+
+const packageInfo = require('../package.json')
+const shrinkwrap = require('../npm-shrinkwrap.json')
+
+/**
+ * @summary Delete a dependency
+ *
+ * @param {*} keys - Keys reference
+ * @param {*} tree - Tree to iterate over
+ * @param {*} onNode - Function to apply
+ * @param {*} parentName - Name of dependent
+ * @param {*} depType - Dependency type (dependencies|requires)
+ *
+ * @example
+ * removeDependency(keys, tree, onNode, parentName)
+ */
+const removeDependency = (keys, tree, onNode, parentName, depType) => {
+  let name = null
+  for (let index = 0; index < keys.length; index += 1) {
+    name = keys[index]
+
+    // Check for this dependency to still exist,
+    // as a node might have been removed just before this iteration
+    if (tree[depType][name]) {
+      onNode(tree, parentName || tree.name, name, tree[depType][name], depType)
+
+      // Check that the walking function didn't remove the dependency
+      // if so, skip it and continue with the next one
+      if (tree[depType][name] && tree[depType][name][depType]) {
+        traverseDeps(tree[depType][name], onNode, name)
+      }
+    }
+  }
+}
+
+/**
+ * @summary Traverse a shrinkwrap tree and call
+ * a given function on each dependency
+ * @param {Object} tree - shrinkwrap
+ * @param {Function} onNode - callback({Object} parent, {String} parentName, {String} name, {Object} info)
```

Comment:
I'm a bit confused... the `onNode` callback in the example on line 70 and actually used on line 91 seems to have 3 parameters, the documented prototype for `onNode` on line 67 has 4 parameters, and the `onNode` usage on line 52 is passing in 5 parameters ???